mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -82,15 +82,18 @@ where
         // TODO: optimize for the common case of a single chunk.
         let mut buffer = BytesMut::default();
         while let Some(chunk) = result.next().await {
-            let (offset, body) = chunk?;
-            if offset != buffer.len() as u64 {
-                return Err(DataCacheError::InvalidBlockOffset);
+            match chunk {
+                Ok((offset, body)) => {
+                    if offset != buffer.len() as u64 {
+                        return Err(DataCacheError::InvalidBlockOffset);
+                    }
+                    buffer.extend_from_slice(&body);
+
+                    result.as_mut().increment_read_window(self.block_size as usize);
```

Comment:
Needs a comment. I already cannot tell anymore whether it is okay to increment **after** calling `next()` for example.